sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTableSchemaInferrer.scala

```diff
@@ -98,6 +99,13 @@ private object CosmosTableSchemaInferrer
       queryOptions.setMaxBufferedItemCount(cosmosInferenceConfig.inferSchemaSamplingSize)
       val queryText = cosmosInferenceConfig.inferSchemaQuery match {
         case None =>
+          ImplementationBridgeHelpers
+            .CosmosQueryRequestOptionsHelper
+            .getCosmosQueryRequestOptionsAccessor
+            .disallowQueryPlanRetrieval(queryOptions)
```

Comment:
The cosmosReadConfig.customQuery could also have a TOP, right? And the default query `SELECT TOP X * from c` also has TOP, they might need the query plan?

Reply:
No - for custom query this isn't an issue. Because Spark even today would issue a separate query for each feed range - so the top would always be local to that feed range (when scoping a query to a specific feed range the query plan will only ever look at that feed range). So none of the challenges with distinct, TOP, aggregates and ordering are an issue because the way hoe Spark queries are executed they would always create partial results for individual feed ranges - if you expect any global ordering, global top etc. you'll have to make additional transformations via Spark on top of whatever the database/connector returned. Schema inference is different - we issue just one global query without feed range scope - there you could break scenarios where custom query for schema inference is used without query plan.

Reply:
My point is, we are doing this if `cosmosInferenceConfig.inferSchemaQuery` is `None` but a custom query can also come from down below with `cosmosReadConfig.customQuery`, so if we are adding this, shouldn't it be on line 110 where both custom queries are `None` and we are defaulting to the default TOP query?

Reply:
Technically you are right - practically I don't think that matches most customer's intention. CustomQuery was very very commonly used for Spark 2.4 - it often contains projections (only couple columns selected etc.) - so when enabling schema inference and having a custom query and knowing that the custom query will only ever be executed in the scope of feedranges/partitions i find it hard to believe that customers would expect any aggregates/ordering/top x to be applied globally during schema inference. Again technically you are right - this change could break some odd edge case. But there would be an easy workaround - use the same query you use as custom query as the schema inference query. So I would rather use the model that matches most customers (avoiding the perf penalty for query plan when that is useless anyway when the query gets executed.